src/v/redpanda/admin_server.cc

```diff
@@ -531,8 +540,9 @@ ss::future<> admin_server::throw_on_error(
 }
 
 void admin_server::register_config_routes() {
-    static ss::httpd::handle_function get_config_handler =
-      []([[maybe_unused]] ss::const_req req, ss::reply& reply) {
+    thread_local static ss::httpd::handle_function get_config_handler =
+      [this](ss::const_req req, ss::reply& reply) {
+          _auth.authenticate(req).require_superuser();
```

Comment:
is this handler (and `get_node_config_handler`) not compatible with the `register_route` helper?

Reply:
They're not.  They're special because they return strings rather than json-serializable objects, which is a whole different callback style in seastar.

However, you've nudged me to create a separate variant of the route registration hook that handles these functions, so that they're slightly less special+different.

Reply:
Cool thanks for the explanation